### ISSUE TITLE
*test-receipts.test receipt update* error binding to port 8099

### ISSUE THREAD

**Issue by mykmelez** [mozilla/r2d2b2g#641]
If I `make test` more than once in a row, then _test-receipts.test receipt update_ fails the second and subsequent times:

```
info: r2d2b2g: executing 'r2d2b2g/tests/test-receipts.test receipt update'
error: r2d2b2g: fail: Error binding to port 8099, did you forget to call simulator.unload? - true == false
info: r2d2b2g: Traceback (most recent call last):
  File "resource://gre/modules/commonjs/sdk/timers.js", line 31, in notify
    callback.apply(null, args);
  File "resource://gre/modules/commonjs/sdk/deprecated/unit-test.js", line 288, in null
    timer.setTimeout(function() { onDone(self); }, 0);
  File "resource://gre/modules/commonjs/sdk/deprecated/unit-test.js", line 458, in null
    options.onDone(self);
  File "resource://gre/modules/commonjs/sdk/deprecated/unit-test.js", line 440, in runNextTest
    self.start({test: test, onDone: runNextTest});
  File "resource://gre/modules/commonjs/sdk/deprecated/unit-test.js", line 471, in start
    this.test.testFunction(this);
  File "resource://gre/modules/commonjs/sdk/test.js", line 64, in null
    test(assert, function() {
  File "resource://r2d2b2g-at-mozilla-dot-org/r2d2b2g/tests/test-receipts.js", line 88, in receiptUpdate
    assert.ok(false, "Error binding to port 8099, did you forget to call " +
  File "resource://gre/modules/commonjs/sdk/test/assert.js", line 103, in ok
    operator: "=="
  File "resource://gre/modules/commonjs/sdk/test/assert.js", line 89, in fail
    this._log.fail(message);
  File "resource://gre/modules/commonjs/sdk/deprecated/unit-test.js", line 88, in fail
    this.console.trace();
  File "resource://gre/modules/commonjs/sdk/lang/functional.js", line 71, in null
    return function() fn.apply(this, args.concat(Array.slice(arguments)));
```

@nickdesaulniers any idea what's going wrong here? I don't remember seeing this when we first landed the receipt tests, although it's possible that I missed it.

**Comment by nickdesaulniers** [comment]
adb is not being shut down properly

**Comment by nickdesaulniers** [comment]
`lsof -i -P` to see it still bound to a port

**Comment by nickdesaulniers** [comment]
WFM from master, did you kill the tests mid run at some point by any chance?

**Comment by mykmelez** [comment]
I might have done so, but I can reproduce consistently by killing adb and then re-running tests, after which an adb process will remain alive and hold onto that port.

I wonder if this has anything to do with the enhancement @bkase landed to leave adb running if it was already running when the Simulator starts it. Perhaps it leaves adb running sometimes even when it was the one to start it.

**Comment by bkase** [comment]
Yeah, this test hit an edge-case in my enhancement.

It looks like the code in `adb.js` is being executed twice when the unit tests start-up.
Look at the top of the logs:

```
Running tests on Firefox 22.0/Gecko 22.0 ({ec8030f7-c20a-464f-9b0e-13a3a9e97384}) under darwin/x86.
info: r2d2b2g: adb: init
info: r2d2b2g: adb: init
```

The flag checking for `adb` running is being reset to false the second time it's executed.

Is it normal for this code to be executing twice?

**Comment by bkase** [comment]
It's not obvious to me why `adb.js` is being executed twice.
Also, I can't seem to easily get a fix working by just changing `adb.js`.

Should keep working around the loading twice? Or does the bug have something to do with the double execution?

**Comment by nickdesaulniers** [comment]
Fixed in d3e7c0cdb29157628b0a01355e34bba3836b6607
